packages/flutter/lib/src/widgets/drag_target.dart

```diff
@@ -316,6 +316,8 @@ class _DragTargetState<T> extends State<DragTarget<T>> {
 
   void didLeave(dynamic data) {
     assert(_candidateData.contains(data) || _rejectedData.contains(data));
+    if (!mounted)
+      return;
```

Comment:
That sounds right to me. The target's internal state will be dropped at dipose time.